pkg/controller/clusterprovision/clusterprovision_controller.go

```diff
@@ -303,9 +333,42 @@ func (r *ReconcileClusterProvision) reconcileRunningJob(instance *hivev1.Cluster
 		}
 	}
 
+	if time.Since(job.CreationTimestamp.Time) < podStatusCheckDelay {
+		timeUntilNextPodStatusCheck := job.CreationTimestamp.Time.Add(podStatusCheckDelay)
+		return reconcile.Result{RequeueAfter: time.Until(timeUntilNextPodStatusCheck)}, nil
+	}
```

Comment:
I would prefer if the values from the if check are used in determining the requeue after, to give it all a bit more cohesion. But it is acceptable as you have it, too.

```suggestion
	if timeUntilNextPodStatusCheck := podStatusCheckDelay - time.Since(job.CreationTimestamp.Time); timeUntilNextPodStatusCheck > 0 {
		return reconcile.Result{RequeueAfter: time.Until(timeUntilNextPodStatusCheck)}, nil
	}
```

```suggestion
	if timeSinceJobCreation := time.Since(job.CreationTimestamp.Time); timeSinceJobCreation < podStatusCheckDelay {
		return reconcile.Result{RequeueAfter: podStatusCheckDelay - timeSinceJobCreation}, nil
	}
```